src/main/java/org/opentripplanner/routing/edgetype/StreetTransitEntityLink.java

```diff
@@ -86,23 +85,38 @@ public State traverse(State s0) {
             return null;
         }
 
-        if (s0.getOptions().bikeParkAndRide && !s0.isBikeParked()) {
-            // Forbid taking your own bike in the station if bike P+R activated.
-            return null;
-        }
-
-        // Do not check here whether any transit modes are selected. A check for the presence of
-        // transit modes will instead be done in the following PreBoard edge.
-        // This allows searching for nearby transit stops using walk-only options.
         StateEditor s1 = s0.edit(this);
 
-        if (s0.getNonTransitMode() == TraverseMode.CAR) {
-            // For Kiss & Ride allow dropping of the passenger before entering the station
-            if (canDropOffAfterDriving(s0) && isLeavingStreetNetwork(req)) {
-                dropOffAfterDriving(s0, s1);
-            } else if (s0.getCarPickupState() != null) {
+        switch (s0.getNonTransitMode()) {
```

Comment:
I generally like to avoid switch statements because of their fall-through by default behavior, even though chained if statements are less readable. (Not insisting on changing this, just a style point.)